### PR TITLE
Improve documentation of robustness test adding key concepts used in the test

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @fuweid @henrybear327 @siyuanfoundation @MadhavJivrajani @jmhbnz

### CODE REVIEW

**Review by fuweid — APPROVED** [review]

File: tests/robustness/README.md [hunk 1]
```diff
@@ -1,17 +1,72 @@
...
+* **Failures:** Network partitions, node crashes, disk failures, and other disruptions.
+
+**Test Procedure:**
+
+1. **Cluster Creation:**  A new etcd cluster is created with the specified configuration.
+2. **Traffic and Failures:** Client traffic is generated and sent to the cluster while failures are injected.
+3. **History Collection:** All client operations and their results are recorded.
+4. **Validation:** The collected history is validated against the etcd model and a set of validators to ensure consistency and correctness.
+5. **Report Generation:**  If a failure is detected, a detailed report is generated to help diagnose the issue.
+   This report includes information about the client operations, etcd data directories.
```
Comment (henrybear327) [inline tests/robustness/README.md:25]:
```suggestion
   This report includes information about the client operations and etcd data directories.
```

**Review by henrybear327 — APPROVED** [review]
LGTM! Thanks @serathius ! :)

File: tests/robustness/README.md [hunk 2]
```diff
@@ -1,17 +1,72 @@
...
+    *   **Revision:** A unique, increasing number assigned to each change in the key-value store, serving as a logical clock.
+    *   **Linearizability (Default):** From the client's perspective, operations appear to happen instantly at some point between their invocation and response.
+    *   **Serializable Consistency (Optional):** Clients can configure read requests to be "serializable" for lower latency, potentially accessing stale data but improving performance.
+*   **Watch API guarantees**
+    *   **Ordered:** Watch events are delivered in order of their revision.
+    *   **Unique:** Each event is delivered only once.
+    *   **Reliable:** No subsequence of events within the available history window will be dropped.
+    *   **Atomic:** Events within the same revision are not split across multiple watch responses.
+    *   **Resumable:** A broken watch can be resumed from the last received revision.
+    *   **Bookmarkable:** Progress notifications guarantee that all events up to a certain revision have been delivered.
```
[the remark this answers is not in the document]
Comment (serathius) [reply tests/robustness/README.md:59]: Replaced with links to https://etcd.io/docs/latest/learning/api_guarantees/

**Pull request merged**